geode-core/src/main/java/org/apache/geode/internal/security/IntegratedSecurityService.java

```diff
@@ -68,9 +69,9 @@
   IntegratedSecurityService(SecurityManagerProvider provider, PostProcessor postProcessor) {
     // provider must provide a shiro security manager, otherwise, this is not integrated security
     // service at all.
-    assert provider.getShiroSecurityManager() != null;
-    SecurityUtils.setSecurityManager(provider.getShiroSecurityManager());
-
+    this.shiroSecurityManager = provider.getShiroSecurityManager();
+    assert this.shiroSecurityManager != null;
+    ThreadContext.bind(this.shiroSecurityManager);
```

Comment:
Also, this will only set the securitymanger in the running thread's ThreadContext, other threads who is handling authentication/authorization won't see this security manager. It will result in more such problems.

Reply:
Also, the question of it being set for other threads is already addressed in in ThreadContext itself by using an InheritableThreadLocal.  We also establish the security manager in getSubject() for corner cases where the thread is spawned by some other mechanism like JMX.